detox/src/logger/utils/streams/DetoxJSONLParser.js

```diff
@@ -0,0 +1,31 @@
+const JsonlParser = require('stream-json/jsonl/Parser');
+
+const { through } = require('./transformers');
+
+class DetoxJSONLParser {
+  constructor(log) {
+    this._log = log;
+  }
+
+  createTransformer() {
+    const log = this._log;
+    const readable = through();
+    const writable = JsonlParser.make({ checkErrors: true })
+      .on('error', (err) => {
+        /* istanbul ignore else */
+        if (err instanceof SyntaxError) {
+          log.debug({ err });
```

Comment:
So this will not be parsed and transformed, obviously. But will the logging be successful? Have you managed to test that out?

Reply:
Yes, that can potentially happen on abruptly terminated log files. I tested this case manually and in integration tests.